src/check/constrain/unify/link.rs

```diff
@@ -28,30 +27,31 @@ pub fn unify_link(constraints: &mut Constraints, finished: &mut Finished, ctx: &
         trace!("{:width$}[{}{}]  {}", pos, unify, msg, constraint, width = 27);
 
         if let Type { name } = &left.expect {
-            finished.push_ty(ctx, right.pos, name)?;
+            if !name.contains_temp() { finished.push_ty(ctx, right.pos, name)?; }
         }
         if let Type { name } = &right.expect {
```

Comment:
These two if statements should not be necessary in theory.
We should only add to finished in a successful union of two types.

Reply:
Remove it instead in ty